src/Iter.mo

```diff
@@ -221,8 +252,8 @@ module {
   /// Like `fromArray` but for Arrays with mutable elements. Captures
   /// the elements of the Array at the time the iterator is created, so
   /// further modifications won't be reflected in the iterator.
-  public func fromVarArray<T>(xs : [var T]) : Iter<T> {
-    fromArray<T>(Array.fromVarArray<T>(xs))
+  public func fromVarArray<T>(iter : [var T]) : Iter<T> {
+    fromArray<T>(Array.fromVarArray<T>(iter))
```

Comment:
```suggestion
    fromArray<T>(Array.fromVarArray<T>(array))
```
Is copying the array intentional? To provide a safe snapshot, I guess?

Reply:
I think I just did this for conciseness, but it definitely seems worth the repeated code to remove the performance overhead of `fromVarArray()`.